internal/ingress/controller/controller.go

```diff
@@ -129,6 +130,21 @@ type Configuration struct {
 	DeepInspector         bool
 
 	DynamicConfigurationRetries int
+
+	EnableTopologyAwareRouting bool
+}
+
+func getIngressPodZone(svc *apiv1.Service) string {
+	svcKey := k8s.MetaNamespaceKey(svc)
+	if svcZoneAnnotation, ok := svc.ObjectMeta.GetAnnotations()[apiv1.AnnotationTopologyAwareHints]; ok {
+		if strings.ToLower(svcZoneAnnotation) == "auto" {
+			if foundZone, ok := k8s.IngressNodeDetails.GetLabels()[apiv1.LabelTopologyZone]; ok {
```

Comment:
just a safety measure (I know you define this, but I'm scared of random panics), check before if the k8s.IngressNodeDetails is not nil :)

Reply:
and what about svc.ObjectMeta ... whis would really never happen till somebody not remove definition and in that case it will not compile.